packages/next/telemetry/events/build.ts

```diff
@@ -19,13 +19,30 @@ type EventBuildOptimized = {
   totalPageCount: number
   staticPageCount: number
   ssrPageCount: number
+  hasDunderPages: boolean
+  hasTestPages: boolean
 }
 
+const DUNDER_PAGES = /^[\\/]__generated__[\\/]/
```

Comment:
We should detect all dunder page directories IMO.

```suggestion
const REGEXP_DIRECTORY_DUNDER = /[\\/]__[^\\/]+(?<!tests)__[\\/]/
```